services/server/server.go

```diff
@@ -522,6 +522,13 @@ func (s *Server) validateChangeSettingsRequest(ctx context.Context, req *serverp
 
 	// check request parameters compatibility with environment variables
 
+	if req.DisableUpdates && s.envSettings.EnableUpdates {
+		return status.Error(codes.FailedPrecondition, "Updates are enabled via ENABLE_UPDATES environment variable.")
+	}
+	if req.DisableUpdates && s.envSettings.EnableTelemetry {
+		return status.Error(codes.FailedPrecondition, "Updates cannot be disabled because telemetry is enabled via ENABLE_TELEMETRY environment variable.")
+	}
+
```

Comment:
user also should not be able to enable updates if it was disabled via env variable.